libsrc/common/strnlen.s

```diff
@@ -0,0 +1,54 @@
+; size_t __fastcall__ strnlen (const char* s, size_t maxlen);
+
+.export     _strnlen
+.import     popax
+.importzp   ptr1, tmp1, tmp2, tmp3, tmp4
+
+.proc _strnlen
+        ; Fetch string pointer.
+        sta     ptr1
+        stx     ptr1+1
+
+        ; Clear return value.
+        lda     #0
+        sta     tmp1
+        sta     tmp2
+
+        ; Get maximum length.
+        jsr     popax
+        sta     tmp3
+        inc     tmp3
+        inx
+        stx     tmp4
+
+        ;;; Loop over string.
+        ldy     #0
```

Comment:
You can save a couple of bytes by using this .Y value to clear the function's return value.